src/PAModel/IR/IRStateHelpers.cs

```diff
@@ -337,15 +337,15 @@ private static (ControlInfoJson.Item item, int index) CombineIRAndState(BlockNod
                     ExtensionData = state.ExtensionData,
                     IsGroupControl = state.IsGroupControl,
                     GroupedControlsKey = state.GroupedControlsKey,
-                    AllowAccessToGlobals = (state.IsComponentDefinition ?? false) ? templateState?.ComponentManifest?.AllowAccessToGlobals : state.AllowAccessToGlobals,
+                    AllowAccessToGlobals = state.AllowAccessToGlobals,
```

Comment:
>state.AllowAccessToGlobals [](http://example.com/codeflow?start=43&length=26)

Is templateState null for component instances?

Ideally we should always pull this from
templateState?.ComponentManifest?.AllowAccessToGlobals

Is this available in non-compdef scenarios?

Reply:
I believe in this case templateState is not null & `templateState.ComponentManifest.AllowAccessToGlobals` doesn't match `state.AllowAccessToGlobals`. For whatever reason, it appears that the `state.AllowAccessToGlobals` is actually the correct value since it is coming directly out of the JSON. If we use the templateState, then it will switch the TopParent.AllowAccessToGlobals to the wrong value.

Reply:
Yup, pulling the property values always from the instances lets us avoid this out of sync situation.